### PR TITLE
Add FlightSQL module docs and links to `arrow-flight` crates

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

re https://github.com/apache/arrow-rs/pull/3887

# Rationale for this change
 
While waiting for CI to run on https://github.com/apache/arrow-rs/pull/3887 I spent some time writing docs to make it easier to understand what is going on / what we have in this crate. 

# What changes are included in this PR?

1. Add a breadcrumb to `arrow-flight` to point at the `sql` feature. 
2. Add module documentation to arrow-flight sql with some orienting text and pointers

# Are there any user-facing changes?

more docs

### CODE REVIEW

File: arrow-flight/src/lib.rs [hunk 1]
```diff
@@ -30,6 +30,11 @@
 //!
 //! 2. Low level [tonic] generated [`flight_service_client`] and
 //! [`flight_service_server`].
+//!
+//! 3. Experimental support for [Flight SQL] in [`sql`]. Requires the
```
Comment (avantgardnerio) [inline arrow-flight/src/lib.rs:34]: I'd love to make this not experimental BTW. It's just a struct and impl in a library AFAIK, so it shouldn't hurt anyone.

**Review by avantgardnerio — APPROVED** [review]
Unquestionably an improvement, ty :)

File: arrow-flight/src/sql/mod.rs [hunk 2]
```diff
@@ -15,6 +15,26 @@
 // specific language governing permissions and limitations
 // under the License.
 
+//! Support for execute SQL queries using [Apache Arrow] [Flight SQL].
+//!
+//! [Flight SQL] is built on top of Arrow Flight RPC framework, by
+//! defining specific message, encoded using the protobuf format, sent
```
Comment (tustvold) [inline arrow-flight/src/sql/mod.rs:21]:
```suggestion
//! defining specific messages, encoded using the protobuf format, sent
```

File: arrow-flight/src/lib.rs (hunk 1 above)
Comment (alamb) [reply arrow-flight/src/lib.rs:34]: Yeah, I think we were just trying to give people the heads up that we would likely be making substantial changes to the API -- but then we do that with other parts of arrow, so maybe the extra "experimental" part is unecessary.

What do you think @tustvold  / @viirya  -- should we rename the feature flag from experimental (and maybe turn it on by default?)

File: arrow-flight/src/sql/mod.rs [hunk 3]
```diff
@@ -15,6 +15,26 @@
 // specific language governing permissions and limitations
 // under the License.
 
+//! Support for execute SQL queries using [Apache Arrow] [Flight SQL].
+//!
+//! [Flight SQL] is built on top of Arrow Flight RPC framework, by
```
[the remark this answers is not in the document]
Comment (alamb) [reply arrow-flight/src/sql/mod.rs:20]: Good idea -- done in 7c8f3e172

File: arrow-flight/src/lib.rs (hunk 1 above)
Comment (tustvold) [reply arrow-flight/src/lib.rs:34]: Where did we get to with an integration test of FlightSQL? I think that would help justify graduating this functionality perhaps? 

I'm somewhat apprehensive there may still be areas we have interpreted the specification differently, one was fixed last week. The experimental flag allows us to "fix" such things without needing to preserve backwards compatibility

Comment (alamb) [reply arrow-flight/src/lib.rs:34]: > The experimental flag allows us to "fix" such things without needing to preserve backwards compatibility

Well, I guess I was thinking we break backwards compatibility fairly regularly (as in there are several API changes per release)

So it isn't like I think the flight sql feature is fully ready and the API won't change -- more like I wonder if we should treat it specially from the rest of arrow-flight

Comment (alamb) [reply arrow-flight/src/lib.rs:34]: I see -- you are perhaps imagining that the FlightSQL protocol itself may change, not just the rust implementation. The spec is marked as experimental in many places. 🤷 

https://github.com/apache/arrow/blob/ad115be1214b13ce393537bdb9c34ae919e4997f/format/FlightSql.proto#LL46

**Pull request merged**